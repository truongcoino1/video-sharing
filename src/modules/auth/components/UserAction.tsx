import React, { memo } from 'react';
import Link from 'next/link';
import {useAuth} from "../hooks/useAuth"

const PATH_SHARE = '/share';

type UserActionProps = {
  email?: string;
};
const UserAction = ({email}: UserActionProps) => {
  const {logout} = useAuth();

  return (
    <div className="user-action-container">
      <span className="txt-welcome">Welcome {email}</span>
      <button
        className="btn-share-movie ml-12"
        disabled={window.location.pathname === PATH_SHARE}
      >
        {window.location.pathname === PATH_SHARE ? (
          'Share a movie'
        ) : (
          <Link href={PATH_SHARE}>Share a movie</Link>
        )}
      </button>
      <button className="btn-logout ml-12" onClick={logout}>
        Logout
      </button>
    </div>
  );
};

export default memo(UserAction);
