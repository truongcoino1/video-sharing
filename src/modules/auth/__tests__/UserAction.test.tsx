import { render, cleanup, fireEvent } from "@testing-library/react";
import UserAction from "../components/UserAction";

jest.mock("next/navigation", () => ({
  useRouter: () => ({
    push: jest.fn(),
  }),
  usePathname: () => "/",
}));

const logout = jest.fn();
jest.mock("../hooks/useAuth", () => {
  const useAuth = () => {
    return {
      logout: logout
    };
  }

  return { useAuth }
});

describe("UserAction", () => {
  afterEach(cleanup);

  it("UserAction should render without crash", () => {
    const result = render(<UserAction />);
    const elm = result.container.querySelector(".user-action-container");
    expect(elm).toBeInTheDocument();
  });

  it("UserAction should show authorized user's email", () => {
    const email = "email_test@gmail.com";
    const result = render(<UserAction email={email} />);
    const elm = result.container.querySelector(".txt-welcome");
    expect(elm).not.toBeNull();
    expect(elm?.textContent).toBe(`Welcome ${email}`);
  });

  it("UserAction should show empty user email when not passing", () => {
    const result = render(<UserAction />);
    const elm = result.container.querySelector(".txt-welcome");
    expect(elm).not.toBeNull();
    expect(elm?.textContent).toBe(`Welcome `);
  });

  it('It calls start logout on button click', () => {
    const result = render(<UserAction />);
    const btn = result.container.querySelector(".btn-logout") as Element;
    fireEvent.click(btn)
    expect(logout).toBeCalledTimes(1);
  });

  it('should render correct Login', () => {
    const element = render(<UserAction />)
    expect(element).toMatchSnapshot();
  });
});
