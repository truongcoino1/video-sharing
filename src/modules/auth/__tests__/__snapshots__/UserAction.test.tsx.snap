// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAction should render correct Login 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="user-action-container flex items-center"
      >
        <span
          class="txt-welcome "
        >
          Welcome 
          <span
            class="text-primary-250 font-semibold"
          />
        </span>
        <button
          class="btn-share-movie ml-12 w-[120px] h-[36px] flex items-center rounded-medium justify-center relative before:z-0  text-dark-high hover:before:bg-light-100/20 bg-dark-200 before:h-full before:w-full before:absolute before:top-0 before:left-0 before:rounded-medium "
        >
          Share a movie
        </button>
        <button
          class="btn-logout ml-12 border border-gray-300 h-[36px] px-[12px] rounded-medium"
        >
          Logout
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="user-action-container flex items-center"
    >
      <span
        class="txt-welcome "
      >
        Welcome 
        <span
          class="text-primary-250 font-semibold"
        />
      </span>
      <button
        class="btn-share-movie ml-12 w-[120px] h-[36px] flex items-center rounded-medium justify-center relative before:z-0  text-dark-high hover:before:bg-light-100/20 bg-dark-200 before:h-full before:w-full before:absolute before:top-0 before:left-0 before:rounded-medium "
      >
        Share a movie
      </button>
      <button
        class="btn-logout ml-12 border border-gray-300 h-[36px] px-[12px] rounded-medium"
      >
        Logout
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
